core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -1106,6 +1115,12 @@ class Partition(val topicPartition: TopicPartition,
    * @return true if the HW was incremented, and false otherwise.
    */
   private def maybeIncrementLeaderHW(leaderLog: UnifiedLog, currentTimeMs: Long = time.milliseconds): Boolean = {
+    if (metadataCache.isInstanceOf[KRaftMetadataCache] && interBrokerProtocolVersion.isElrSupported && eligibleLeaderReplicasEnabled) {
+      if (isUnderMinIsr) {
+        debug(s"Skip checking whether HWM can advance because partition=$topicPartition is under min ISR(ISR=${partitionState.isr}")
```

Comment:
Let's set this to TRACE. Also, we can omit the `$topicPartition` part since it's included in the logIdent for Partition